source/CIPopupMenu.cpp

```diff
@@ -0,0 +1,245 @@
+                    state = MenuState::MainToLobby;
+                }
+            }
+            else if (_timer > 0.0f) {
+                // display error 
+                _window->setVisible(false);
+                _error->setVisible(true);
+            }
+            else {
+                // return to main menu
+                state = MenuState::LoadToMain;
+            }
+            break;
+        }
+        case MenuState::JoiningGame:
+        {
+            _timer -= timestep;
+
+            // TODO: make this circle rotate properly
+            //_windowCircle->setAngle(_windowCircle->getAngle() + ((timestep / 0.5f) * 6.28319));
+            if (_timer > 11.0f) {
+                _networkMessageManager->sendMessages();
+                _networkMessageManager->receiveMessages();
+                break;
+            }
+            else if (_timer > 2.5f) {
+                _networkMessageManager->sendMessages();
+                _networkMessageManager->receiveMessages();
+
+                if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::RoomNotFound) {
+                    CULog("Invalid room id.");
+                    _timer = 2.4f; // skip to displaying error message
+                    break;
+                }
+                else if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Connected) {
+                    CULog("Successfully joined game.");
+                    state = MenuState::JoinToLobby;
+                }
+            }
+            else if (_timer > 0.0f) {
+                // display error message
+                _window->setVisible(false);
+                _error->setVisible(true);
+            }
+            else {
+                // return to join menu
+                state = MenuState::MainToJoin;
+            }
+            break;
+        }
+        case MenuState::ReconnectingGame:
+        {
+            _timer -= timestep;
+
+            // TODO: make this circle rotate properly
+            //_windowCircle->setAngle(_windowCircle->getAngle() + ((timestep / 0.5f) * 6.28319));
+            if (_timer > 11.0f) {
+                _networkMessageManager->sendMessages();
+                _networkMessageManager->receiveMessages();
+                break;
+            }
+            else if (_timer > 2.5f) {
+                _networkMessageManager->sendMessages();
+                _networkMessageManager->receiveMessages();
+
+                if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Reconnecting) {
+                    break;
+                }
+                else if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Disconnected) {
+                    _timer = 2.4f;
+                    break;
+                }
+                else if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Connected) {
```

Comment:
I think you need a case for CUNetworkConnection::NetStatus::GenericError. This should cause it to not try to recreate a game after failing to create a game